types/relay-runtime/lib/store/RelayStoreTypes.d.ts

```diff
@@ -347,9 +379,11 @@ export interface ReadOnlyRecordSourceProxy {
  * Extends the RecordSourceProxy interface with methods for accessing the root
  * fields of a Selector.
  */
-export interface RecordSourceSelectorProxy extends RecordSourceProxy {
-    getRootField(fieldName: string): RecordProxy | null | undefined;
-    getPluralRootField(fieldName: string): Array<RecordProxy | null | undefined> | null | undefined;
+
+export interface RecordSourceSelectorProxy<T = {}> extends RecordSourceProxy {
+    getRootField<K extends keyof T>(fieldName: K): RecordProxy<NonNullable<T[K]>>;
+    getRootField(fieldName: string): RecordProxy | null;
```

Comment:
@mattkrick @alloy Why did we remove `| undefined` here? It's what the upstream Flow types declare:

https://github.com/facebook/relay/blob/7fd936ac9eddb44b6f463affa03679a53a6d0561/packages/relay-runtime/store/RelayStoreTypes.js#L378

Reply:
Hmm, yep indeed–my bad for forgetting about flow’s `?Foo` details.

Reply:
ah, that's my bad, must've gotten a little too excited.
just outta curiosity, how do you get that to return undefined?? i thought the server spec only allowed null?

Reply:
Oh, I don't know really whether it ever does. It's just marked as such in Flow. I only noticed because the removal of `undefined` here caused me to get type checking failures from different versions of this typing lib being installed.